scripts/create-clusters.sh

```diff
@@ -123,18 +124,21 @@ function fixup-cluster() {
   # TODO(font): Need to set container IP address in order for clusters to reach
   # kube API servers in other clusters until
   # https://github.com/kubernetes-sigs/kind/issues/111 is resolved.
-  local container_ip_addr=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' kind-${i}-control-plane)
-  sed -i "s/localhost/${container_ip_addr}/" ${kubeconfig_path}
-
+  if [ "$OS" != "Darwin" ];then
+      local container_ip_addr=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' kind-${i}-control-plane)
+      sed -i.bak "s/localhost/${container_ip_addr}/" ${kubeconfig_path} && rm -rf ${kubeconfig_path}.bak
+  fi
   # TODO(font): Need to rename auth user name to avoid conflicts when using
   # multiple cluster kubeconfigs. Remove once
   # https://github.com/kubernetes-sigs/kind/issues/112 is resolved.
-  sed -i "s/kubernetes-admin/kubernetes-kind-${i}-admin/" ${kubeconfig_path}
+  # Compatible with MacOS
+  sed -i.bak "s/kubernetes-admin/kubernetes-kind-${i}-admin/" ${kubeconfig_path} && rm -rf ${kubeconfig_path}.bak
 }
 
 function check-clusters-ready() {
   for i in $(seq ${1}); do
-    util::wait-for-condition 'ok' "kubectl --context cluster${i} get --raw=/healthz &> /dev/null" 120
+    local kubeconfig_path="$(kind get kubeconfig-path --name kind-${i})"
+    util::wait-for-condition 'ok' "kubectl --kubeconfig ${kubeconfig_path} --context cluster${i} get --raw=/healthz &> /dev/null" 120
```

Comment:
is this a general bug fix?

Reply:
No, an enhancement. Make it more specific.  Could be removed.

Reply:
I think this is fine.